Meson /  Ninja under MinGW64 mangled output, implicit Werror, etc

Hi there, is there any special guidance for using the meson toolchain under Windows-MinGW64?

First the output is hard to read, probably due to python3 utf8 attempt to colorize its output incorrectly under windows, here attempting hexchat build:
```
meson mingw64 -Dprefix=/MSYS2/mingw64 -Dwith-libcanberra=false -Dwith-perl=false -Dwith-upd=false -Dwith-winamp=false -Dwith-appdata=false -Dwith-libnotify=false -Dwith-lua=false

←[1mThe Meson build system←[0m
Version: 0.49.0
Source dir: ←[1mD:\Download\MinGW-64-souorce\hexchat-2.14.2←[0m
Build dir: ←[1mD:\Download\MinGW-64-souorce\hexchat-2.14.2\mingw64←[0m
Build type: ←[1mnative build←[0m
Project name: ←[1mhexchat←[0m
Project version: ←[1m2.14.2←[0m
Native C compiler: ←[1mcc←[0m (gcc 8.2.0 "cc (Rev3, Built by MSYS2 project) 8.2.
0")
Build machine cpu family: ←[1mx86_64←[0m
Build machine cpu: ←[1mx86_64←[0m
Found pkg-config: ←[1mD:\MSYS2\mingw64\bin\pkg-config.EXE←[0m (0.29.2)
Dependency ←[1mgio-2.0←[0m found: ←[1;32mYES←[0m 2.58.2
...

```

Could it be this is due to python3 specific use of utf8 as sys.defaultencoding under windows?
If I change the console codepage using "chcp 65001", things get a _little_ better(?):
```[01m[K../src/common/text.c:203:26:[m[K [01;31m[Kerror: [m[Kformat '[01m
[K%i[m[K' expects argument of type '[01m[Kint[m[K', but argument 2 has ty
pe '[01m[Klong long int[m[K' [[01;31m[K-Werror=format=[m[K]
   buf = g_strdup_printf ([01;31m[K"T %"[m[K G_GINT64_FORMAT " ", [32m[K(g
int64)stamp[m[K);
                          [01;31m[K^~~~~[m[K                      [32m[K~~
~~~~~~~~~~~[m[K
In file included from [01m[KD:/MSYS2/mingw64/include/glib-2.0/glib/gtypes.h:32
[m[K,
                 from [01m[KD:/MSYS2/mingw64/include/glib-2.0/glib/galloca.h:3
2[m[K,
                 from [01m[KD:/MSYS2/mingw64/include/glib-2.0/glib.h:30[m[K,

                 from [01m[K../src/common/hexchat.h:22[m[K,
                 from [01m[K../src/common/text.c:35[m[K:
[01m[KD:/MSYS2/mingw64/include/glib-2.0/glibconfig.h:62:29:[m[K [01;36m[Kn
ote: [m[Kformat string is defined here
 #define G_GINT64_FORMAT "I64i"
cc1.exe: some warnings being treated as errors
[51/87] Compiling C object plugins/sys...rc_common_sysinfo_win32_backend.c.obj.
ninja: build stopped: subcommand failed.
```

Also notice apparent use of Werror CFLAG, etc. Any guidance on best practice fixing things like this under MinGW64? thanks.

--- a/mingw-w64-meson/PKGBUILD
+++ b/mingw-w64-meson/PKGBUILD
@@ -3,7 +3,7 @@
 _realname=meson
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
-pkgver=0.50.0
+pkgver=0.50.1
 pkgrel=1
 pkgdesc="High-productivity build system (mingw-w64)"
 arch=('any')
@@ -18,8 +18,8 @@ source=("https://github.com/mesonbuild/${_realname}/releases/download/${pkgver}/
         '0002-Default-to-sys.prefix-as-the-default-prefix.patch'
         '0003-Strip-the-prefix-from-all-paths-when-installing-with.patch'
         'install-man.patch')
-sha256sums=('2a1bc42dda58206fb922cda5e1ca95cc03ad126321d26acc47d3493ec4e7021f'
-            'edec6de6f5fa60f0737fdb25a3ac276a0187a6f961544289e5472bcb4194e954'
+sha256sums=('f68f56d60c80a77df8fc08fa1016bc5831605d4717b622c96212573271e14ecc'
+            '5805aed0a117536eb16dd8eef978c6be57c2471b655ede63e25517c28b4f4cf0'
             '4ef1e6fb0e9bd927b4646e416631b8a0a9aece763d259dbba58168bcba93f673'
             '2093c617cf3146a4cea601e7c73400d8ec5fd52ac5cf642c4f5af2d6493b1cb1'
             '0682a36cb75e545a78b81293303835a16171f25baf949905dc08029436efff84')

--- a/mingw-w64-meson/color-term.patch
+++ b/mingw-w64-meson/color-term.patch
@@ -1,11 +1,11 @@
---- meson-0.47.0/mesonbuild/mlog.py.orig	2018-07-03 13:53:50.151713800 +0200
-+++ meson-0.47.0/mesonbuild/mlog.py	2018-07-03 14:15:33.385701800 +0200
-@@ -34,7 +34,7 @@
-     return kernel.SetConsoleMode(stdout, mode.value | 0x4) or os.environ.get('ANSICON')
- 
- if platform.system().lower() == 'windows':
--    colorize_console = os.isatty(sys.stdout.fileno()) and _windows_ansi()
-+    colorize_console = os.isatty(sys.stdout.fileno())
- else:
-     colorize_console = os.isatty(sys.stdout.fileno()) and os.environ.get('TERM') != 'dumb'
- log_dir = None
+--- meson-0.50.1/mesonbuild/mlog.py.orig	2019-03-06 21:48:10.000000000 +0100
++++ meson-0.50.1/mesonbuild/mlog.py	2019-04-17 17:46:10.229092600 +0200
+@@ -31,7 +31,7 @@
+     stdout = kernel.GetStdHandle(-11)
+     mode = DWORD()
+     if not kernel.GetConsoleMode(stdout, byref(mode)):
+-        return False
++        return os.isatty(sys.stdout.fileno())
+     # ENABLE_VIRTUAL_TERMINAL_PROCESSING == 0x4
+     # If the call to enable VT processing fails (returns 0), we fallback to
+     # original behavior

